parcels/kernel.py

```diff
@@ -218,6 +218,9 @@ def load_lib(self):
 
     def execute_jit(self, pset, endtime, dt):
         """Invokes JIT engine to perform the core update loop"""
+        if len(pset) > 0 and pset.particle_data['xi'].ndim == 2:
+            assert pset.fieldset.gridset.size == pset.particle_data['xi'].shape[0], \
+                'FieldSet has different number of grids than Particle.xi. Have you added Fields after creating the ParticleSet?'
```

Comment:
yes, great - I also thought of re-introducing that check from earlier version. Good idea.